cdap/http.go

```diff
@@ -44,13 +44,22 @@ func httpCall(client *http.Client, req *http.Request) ([]byte, error) {
 
 	b, err := doHTTPCall(client, req)
 
-	// CDAP REST intermittently returns 500 internal errors, we will retry on 500s once.
+	// CDAP REST intermittently returns 500, 504, etc. internal errors, we will retry on 5xxs once.
 	var e *httpError
-	if errors.As(err, &e) && e.code == 500 {
-		log.Print("retrying on intermittent 500 error in 2 seconds")
-		time.Sleep(2 * time.Second)
-		b, err = doHTTPCall(client, req)
-	}
+    // poor man's 3x EBO.
+    s := 2
+    for i:=0; i < 3; i++ {
+        if errors.As(err, &e) && e.code >= 500 && e.code < 600 {
+            log.Printf("retrying on intermittent 5xx error in %v seconds", s)
+            // Have to explicitly cast if int var: https://play.golang.org/d0ZFLSVoAw
+            time.Sleep(time.Duration(s) * time.Second)
+            s *= 2
```

Comment:
time.Sleep(time.Duration(2*i) * time.Second) ? no ned for s anymore

Reply:
This would be linear not exponential backoff.

Reply:
I will use math.Pow(i, 2)